src/py/pyodide/ffi/wrappers.py

```diff
@@ -38,7 +50,8 @@ def add_event_listener(
     JsProxy corresponding to the ``listener`` parameter.
     """
     proxy = create_proxy(listener)
-    EVENT_LISTENERS[(elt.js_id, event, listener)] = proxy
+    # Use weakref so the proxy can be freed if DOM element is removed.
```

Comment:
FYI listeners attached to elements don't leak and are free to go once the element is GCd

Reply:
Prior to this change, ones added via this `add_event_listener()` method would leak because they would stay in this `EVENT_LISTENERS` map forever.

Reply:
I see ... then this makes sense ... **but** ... if only there was a *FinalizationRegistry* at the *JS* side of affairs that would help avoiding entirely these issue with proxies ;-)

I hope one day proxies passed to JS will be observed at the GC level by default, can't wait for that WASM GC hook that helps with double links to land and be used in here too.

Reply:
There is a finalization registry but if we hold a strong reference to a JavaScript object it will never be GCd and the finalization registry callback will never be called. What's needed to allow the finalization registry to work is that we hold a weak ref here.

Reply:
FYI I've elaborated further my idea of a *WeakProxy* in here: https://github.com/WebReflection/weak-proxy

feel free to grab that code and change it if it helps making everything less `create_proxy` / `create_proxy_once` dependent 👋